pattern_importer.py

```
@@ -0,0 +1,322 @@
+            # Check cache age
+            cache_age = time.time() - self.CACHE_FILE.stat().st_mtime
+            if cache_age > self.CACHE_EXPIRY_HOURS * 3600:
+                logger.info("Pattern cache expired")
+                return None
+            
+            with open(self.CACHE_FILE, 'r') as f:
+                patterns = json.load(f)
+            
+            return patterns
+            
+        except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)